internal/cloud/aws/aws.go

```diff
@@ -81,7 +79,12 @@ func New(ctx context.Context) (*Cloud, error) {
 
 // List retrieves all instances belonging to the current Constellation.
 func (c *Cloud) List(ctx context.Context) ([]metadata.InstanceMetadata, error) {
-	uid, err := readInstanceTag(ctx, c.imds, cloud.TagUID)
+	identity, err := c.imds.GetInstanceIdentityDocument(ctx, &imds.GetInstanceIdentityDocumentInput{})
+	if err != nil {
+		return nil, fmt.Errorf("retrieving instance identity: %w", err)
+	}
+
+	uid, err := c.readInstanceTag(ctx, identity.InstanceID, cloud.TagUID)
```

Comment:
Since all calls to `c.readInstanceTag` require a call to `c.imds.GetInstanceIdentityDocument` first, why not move the identity document fetching into `c.readInstanceTag`?

Reply:
Good point